### ISSUE TITLE
KeyError: 'started_at' when loading trace to the destination (second and consecutive runs)

### ISSUE THREAD

**Issue by a13xa1v35** [dlt-hub/dlt#1584]
### dlt version

0.4.12

### Describe the problem

An exception occurs when loading the pipeline trace to the destination as per the [docs](https://dlthub.com/docs/running-in-production/running#inspect-and-save-the-load-info-and-trace) in the second and consecutive runs (the first load is successful).

```
2024-07-03 12:54:19,832|[INFO                 ]|53157|8499727360|dlt|pool_runner.py|run_pool:104|Processing pool closed
2024-07-03 12:54:19,834|[INFO                 ]|53157|8499727360|dlt|paddle_pipeline.py|run:88|Running _trace - id: 45cfcaf1a766fbd599028a56f948a412 - endpoint: transactions 
KeyError: 'started_at'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/pipeline/pipeline.py", line 431, in extract
    self._extract_source(extract_step, source, max_parallel_items, workers)
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/pipeline/pipeline.py", line 1106, in _extract_source
    load_id = extract.extract(source, max_parallel_items, workers)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/extract/extract.py", line 397, in extract
    self._extract_single_source(
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/extract/extract.py", line 316, in _extract_single_source
    with self.manage_writers(load_id, source):
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/contextlib.py", line 144, in __exit__
    next(self.gen)
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/extract/extract.py", line 355, in manage_writers
    self.extract_storage.close_writers(load_id)
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/extract/storage.py", line 77, in close_writers
    storage.close_writers(load_id, skip_flush=skip_flush)
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/common/storages/data_item_storage.py", line 83, in close_writers
    writer.close(skip_flush=skip_flush)
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/common/data_writers/buffered.py", line 172, in close
    self._flush_and_close_file(skip_flush=skip_flush)
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/common/data_writers/buffered.py", line 226, in _flush_and_close_file
    self._flush_items(allow_empty_file)
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/common/data_writers/buffered.py", line 216, in _flush_items
    self._writer.write_data(self._buffered_items)
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/common/data_writers/writers.py", line 175, in write_data
    json.typed_dump(rows, self._f)
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/common/json/_orjson.py", line 26, in typed_dump
    fp.write(typed_dumpb(obj, pretty=pretty))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/common/json/_orjson.py", line 30, in typed_dumpb
    return _dumps(obj, sort_keys, pretty, custom_pua_encode, orjson.OPT_PASSTHROUGH_DATETIME)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/common/json/_orjson.py", line 18, in _dumps
    return orjson.dumps(obj, default=default, option=options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Type is not JSON serializable: PipelineTrace

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/alexalves/Documents/Boopos/boopos-dlthub/pipelines/paddle_pipeline.py", line 116, in <module>
    pipe.run()
  File "/Users/alexalves/Documents/Boopos/boopos-dlthub/pipelines/paddle_pipeline.py", line 89, in run
    self.pipeline.run([trace], table_name="_trace", loader_file_format="jsonl")
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/pipeline/pipeline.py", line 222, in _wrap
    step_info = f(self, *args, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/pipeline/pipeline.py", line 267, in _wrap
    return f(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/pipeline/pipeline.py", line 674, in run
    self.extract(
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/pipeline/pipeline.py", line 222, in _wrap
    step_info = f(self, *args, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/pipeline/pipeline.py", line 176, in _wrap
    rv = f(self, *args, **kwargs)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/pipeline/pipeline.py", line 162, in _wrap
    return f(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/pipeline/pipeline.py", line 267, in _wrap
    return f(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alexalves/.pyenv/versions/3.11.2/lib/python3.11/site-packages/dlt/pipeline/pipeline.py", line 446, in extract
    raise PipelineStepFailed(
dlt.pipeline.exceptions.PipelineStepFailed: Pipeline execution failed at stage extract when processing package 1720004059.891855 with exception:

<class 'TypeError'>
Type is not JSON serializable: PipelineTrace
```

### Expected behavior

Load the trace successfully as in the first run.

### Steps to reproduce

- Create an incremental pipeline with API as the source, BigQuery as a destination, and GCS as staging.
- Run the pipeline for the first time.
- Load the trace to the destination for the first time.
- Run the pipeline for the second time.
- Load the trace to the destination.
- Exception.

### Operating system

macOS

### Runtime environment

Local

### Python version

3.11

### dlt data source

API

### dlt destination

Google BigQuery

### Additional information

The same error occurs if, instead of loading the trace, you just call `trace.asdict()`

**Comment by rudolfix** [comment]
@sh-rp pls check that out. we have trace serialization tests but we do not check double load. why there's exception after calling `trace.asdict()` is a little bit weird...

**Comment by sh-rp** [comment]
will do

**Comment by VioletM** [comment]
@sh-rp also discussion in Slack: https://dlthub-community.slack.com/archives/C04DQA7JJN6/p1720004615084119

**Comment by rudolfix** [comment]
@a13xa1v35  I fixed this bug in #1641 

@sh-rp I assigned you as reviewer
